python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py

```diff
@@ -0,0 +1,200 @@
+from semantic_kernel.data.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.exceptions import MemoryConnectorException, MemoryConnectorInitializationError
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+TModel = TypeVar("TModel")
+
+
+@experimental_class
+class AzureAISearchCollection(VectorStoreRecordCollection[str, TModel], Generic[TModel]):
+    search_client: SearchClient
+    search_index_client: SearchIndexClient
+    supported_key_types: ClassVar[list[str] | None] = ["str"]
+    supported_vector_types: ClassVar[list[str] | None] = ["float", "int"]
+
+    def __init__(
+        self,
+        data_model_type: type[TModel],
+        data_model_definition: VectorStoreRecordDefinition | None = None,
+        collection_name: str | None = None,
+        search_index_client: SearchIndexClient | None = None,
+        search_client: SearchClient | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initializes a new instance of the AzureAISearchCollection class.
+
+        Args:
+            data_model_type (type[TModel]): The type of the data model.
+            data_model_definition (VectorStoreRecordDefinition): The model definition, optional.
+            collection_name (str): The name of the collection, optional.
+            search_index_client (SearchIndexClient): The search index client for interacting with Azure AI Search,
+                used for creating and deleting indexes.
+            search_client (SearchClient): The search client for interacting with Azure AI Search,
+                used for record operations.
+            **kwargs: Additional keyword arguments, including:
+                The same keyword arguments used for AzureAISearchVectorStore:
+                    search_endpoint: str | None = None,
+                    api_key: str | None = None,
+                    azure_credentials: AzureKeyCredential | None = None,
+                    token_credentials: AsyncTokenCredential | TokenCredential | None = None,
+                    env_file_path: str | None = None,
+                    env_file_encoding: str | None = None
+
+        """
+        if search_client and search_index_client:
+            if not collection_name:
+                collection_name = search_client._index_name
+            elif search_client._index_name != collection_name:
+                raise MemoryConnectorInitializationError(
+                    "Search client and search index client have different index names."
+                )
+            super().__init__(
+                data_model_type=data_model_type,
+                data_model_definition=data_model_definition,
+                collection_name=collection_name,
+                search_client=search_client,
+                search_index_client=search_index_client,
+            )
+            return
```

Comment:
nitpick: could this early `return`s be replaced with `elif`s and a final `else` on line 99 instead?